lib/lita/handlers/puppet.rb

```diff
@@ -136,6 +145,31 @@ def node_profiles(response)
         end
       end
 
+      def nodes_with_class(response)
+        puppet_class = response.matches[3]
+        url = config.puppetdb_url
+        username = friendly_name(response.user.name)
+
+        unless url
+          cant_reply = "#{username}, I would do that, but I don't know how to connect to PuppetDB."
+          cant_reply << "Edit my config and add `config.handlers.puppet.puppetdb_url`."
+          response.reply(cant_reply)
+          return false
+        end
+
+        response.reply("#{username}, let me see what I can find in PuppetDB for you.")
+
+        puppet_classes = class_nodes(url, puppet_class)
+        if puppet_classes.empty?
+          response.reply("There are no nodes with #{puppet_class} class, are you sure its a valid class?")
+          return false
+        else
+          response.reply("Here are all the nodes with class #{puppet_class}:")
+          response.reply("/code" + puppet_class.join("\n"))
```

Comment:
Here I believe you want `puppet_class.join("\n")` to really be `puppet_classes.join("\n")`, since that is the result Array from `class_nodes()`. This might be less confusing if `puppet_classes` were named `puppet_nodes` since that is what is being returned anyway, but that is not super important. As is though, this call would just return the name of the class you're looking for.